File: Chapter 05/Python/section_5_6_crypto_tracker/after/infrastructure/adapters/fake_price_provider.py
from ...core.ports.price_provider_port import PriceProviderPort

class FakePriceProvider(PriceProviderPort):
    """
    ADAPTER 1: The 'Airplane Mode' / Test Adapter.
    """

    def __init__(self, fixed_price: float = 50000.0):
        self.fixed_price = fixed_price

    def get_bitcoin_price(self) -> float:
        return self.fixed_price

File: Chapter 05/JS/section_5_6_crypto_tracker/after/infrastructure/adapters/fakePriceProvider.js
const PriceProviderPort = require('../../core/ports/priceProviderPort');

/**
 * ADAPTER 1: The "Airplane Mode" / Test Adapter.
 */
class FakePriceProvider extends PriceProviderPort {
    constructor(fixedPrice = 50000) {
        super();
        this.fixedPrice = fixedPrice;
    }

    async getBitcoinPrice() {
        return this.fixedPrice;
    }
}

module.exports = FakePriceProvider;